npm.js

```diff
@@ -41,13 +41,21 @@ exports.translate = function(load){
 		npmLoad: npmLoad,
 		crawl: crawl,
 		resavePackageInfo: resavePackageInfo,
-		forwardSlashMap: {}
+		forwardSlashMap: {},
+		// default file structure for npm 3 and higher
+		isFlatFileStructure: true
 	};
 	this.npmContext = context;
 	var pkg = {origFileUrl: load.address, fileUrl: utils.relativeURI(loader.baseURL, load.address)};
 	crawl.processPkgSource(context, pkg, load.source);
-	if(pkg.system && pkg.system.npmAlgorithm === "flat") {
-		context.isFlatFileStructure = true;
+
+	var pkgName = pkg.name;
+	// backwards compatible for < npm 3
+	if(pkg.system && pkg.system.npmAlgorithm === "nested") {
+		context.isFlatFileStructure = false;
+	} else {
+		pkg.system = pkg.system || {};
+		pkg.system.npmAlgorithm = "flat";
```

Comment:
I don't so much care about `System.npmAlgorithm`, it's not something you would use in an application... but I guess we can leave it just in case there's some weird use-case where a user needs to know which npm version is used.